src/fromager/sdist.py

```diff
@@ -579,6 +589,7 @@ def safe_install(
             ":all:",
         ]
         + ctx.pip_wheel_server_args
+        + ctx.pip_constraint_args
```

Comment:
Is this needed for `safe_install`? We already fix the version when calling it from `_maybe_install`: https://github.com/python-wheel-build/fromager/pull/330/files#diff-3f261cb4d56f2826fc58f44967dd7d1b2a92b5e4e9da0133b832d0e9027ff011R564
(and i don't think `resolved_version` is every `None` in that function - we should probably remove that check)

Reply:
OK, I included it here because I did see that conditional check in the caller.

I would prefer to be safe and always use the constraints file. Those rules should be leading us to produce the same version, so it's redundant but still safe.

Reply:
okay fair enough, the e2e tests are failing though not sure why